src/structures/tss.rs

```diff
@@ -37,3 +38,31 @@ impl TaskStateSegment {
         }
     }
 }
+
+/// The given IO permissions bitmap is invalid.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum InvalidIoMap {
+    /// The IO permissions bitmap is too far from the TSS. It must be within `0xdfff` bytes of the
+    /// start of the TSS.
+    TooFarFromTss {
+        /// The distance of the IO permissions bitmap from the beginning of the TSS.
+        distance: usize,
```

Comment:
How much error info should be passed here? The user is able to find all of this out for themselves trivially from the `tss` and `iomap` args, so is it necessary?

Reply:
I agree that it's probably not necessary, but it's nice to have!